runtime/service.go

```diff
@@ -738,3 +754,32 @@ func packBundle(path string, options *ptypes.Any) (*ptypes.Any, error) {
 	}
 	return ptypes.MarshalAny(extraData)
 }
+
+// setupStubDrive will create a flat file with the drive ID as its contents.
+// The file is padded with null bytes to a minimum size of minFCDriveImageSize.
+// This is due to Firecracker minimum drive image size limits.
+func setupStubDrive(builder firecracker.DrivesBuilder, dir, driveID string) (firecracker.DrivesBuilder, error) {
+	path := filepath.Join(dir, driveID)
+	f, err := os.Create(path)
+	if err != nil {
+		return builder, err
+	}
+	defer f.Close()
+
+	if _, err = f.WriteString(driveID); err != nil {
+		return builder, err
+	}
+
+	// Firecracker will not show any drives smaller than 512 bytes. If the drive
+	// size is not a multiple of 512, then those bytes will not be visible to
+	// firecracker.
+	if err := os.Truncate(path, minFCDriveImageBytes); err != nil {
```

Comment:
nit: while it seems unlikely we or someone else would make a change in the future that would result in this, we may as well add a check that the drive ID is less than 512 bytes just in case